src/base/compute/owl_computation_symbol_sig.ml

```diff
@@ -6,211 +6,217 @@
   (** TODO *)
+  val set_reuse : attr Owl_graph.node -> bool -> unit
 
-  val get_reuse : attr Owl_graph.node -> bool
   (** TODO *)
+  val get_reuse : attr Owl_graph.node -> bool
 
-  val is_shared : attr Owl_graph.node -> bool
   (** TODO *)
+  val is_shared : attr Owl_graph.node -> bool
 
-  val get_shared_nodes : attr Owl_graph.node -> (attr Owl_graph.node) array
   (**
   ``get_shared_nodes node`` returns the nodes sharing the same block of memory
   as ``node``.
    *)
+  val get_shared_nodes : attr Owl_graph.node -> attr Owl_graph.node array
 
-  val is_var : attr Owl_graph.node -> bool
   (** TODO *)
+  val is_var : attr Owl_graph.node -> bool
 
-  val is_const : attr Owl_graph.node -> bool
   (** TODO *)
+  val is_const : attr Owl_graph.node -> bool
 
-  val is_node_arr : attr Owl_graph.node -> bool
   (** TODO *)
+  val is_node_arr : attr Owl_graph.node -> bool
 
-  val is_node_elt : attr Owl_graph.node -> bool
   (** TODO *)
+  val is_node_elt : attr Owl_graph.node -> bool
 
-  val is_assigned : attr Owl_graph.node -> bool
   (**
   ``is_assigned node`` checks if a block of memory has been assigned to
   ``node``.
    *)
+  val is_assigned : attr Owl_graph.node -> bool
 
-  val check_assigned : attr Owl_graph.node -> unit
   (**
   ``check_assigned node`` throws an exception if ``node`` has not been
   assigned to a block.
    *)
+  val check_assigned : attr Owl_graph.node -> unit
 
-  val is_valid : attr Owl_graph.node -> bool
   (** TODO *)
+  val is_valid : attr Owl_graph.node -> bool
 
-  val validate : attr Owl_graph.node -> unit
   (** TODO *)
+  val validate : attr Owl_graph.node -> unit
 
-  val invalidate : attr Owl_graph.node -> unit
   (** TODO *)
+  val invalidate : attr Owl_graph.node -> unit
 
-  val invalidate_graph : attr Owl_graph.node -> unit
   (** TODO *)
+  val invalidate_graph : attr Owl_graph.node -> unit
 
-  val is_freeze : attr Owl_graph.node -> bool
   (** TODO *)
+  val is_freeze : attr Owl_graph.node -> bool
 
-  val freeze : attr Owl_graph.node -> unit
   (** TODO *)
+  val freeze : attr Owl_graph.node -> unit
 
-  val freeze_descendants : attr Owl_graph.node array -> unit
   (** TODO *)
+  val freeze_descendants : attr Owl_graph.node array -> unit
 
-  val freeze_ancestors : attr Owl_graph.node array -> unit
   (** TODO *)
+  val freeze_ancestors : attr Owl_graph.node array -> unit
 
-  val pack_arr : A.arr -> arr
   (** TODO *)
+  val pack_arr : A.arr -> arr
 
-  val unpack_arr : arr -> A.arr
   (** TODO *)
+  val unpack_arr : arr -> A.arr
 
-  val pack_elt : A.elt -> elt
   (** TODO *)
+  val pack_elt : A.elt -> elt
 
-  val unpack_elt : elt -> A.elt
   (** TODO *)
+  val unpack_elt : elt -> A.elt
 
-  val unsafe_assign_arr : arr -> A.arr -> unit
   (** TODO *)
+  val unsafe_assign_arr : arr -> A.arr -> unit
 
-  val assign_arr : arr -> A.arr -> unit
   (** TODO *)
+  val assign_arr : arr -> A.arr -> unit
 
-  val assign_elt : elt -> A.elt -> unit
   (** TODO *)
+  val assign_elt : elt -> A.elt -> unit
```

Comment:
oops, i think the comment went back to the top.

Reply:
oh yea I forgot to mention, I've changed owldoc to read the doc with the comments at the above the function and type definitions...
the problem here is even if I set the doc-comments to come after in ocamlformt, it only comes after the functions definitions but not the type definitions (see https://github.com/ocaml-ppx/ocamlformat/issues/556) 

If we really want the comments to come after the function definitions, I can go back to owldoc code, but that will probably take some more time.

Reply:
can ocaml's own doc builder recognise this if the comment is above the function?

Reply:
So that's how I've modified owl doc's `apidoc.ml` this afternoon (see https://github.com/owlbarn/owl_guide/blob/master/bin/apidoc.ml)